airbyte-cdk/python/airbyte_cdk/sources/declarative/requesters/request_options/interpolated_request_body_data_provider.py

```diff
@@ -0,0 +1,34 @@
+#
+# Copyright (c) 2022 Airbyte, Inc., all rights reserved.
+#
+
+from typing import Any, Mapping, Optional, Union
+
+from airbyte_cdk.sources.declarative.interpolation.interpolated_string import InterpolatedString
+from airbyte_cdk.sources.declarative.requesters.interpolated_request_input_provider import InterpolatedRequestInputProvider
+
+
+class InterpolatedRequestBodyDataProvider:
+    """
+    Helper class that allows for performing interpolation on either a string or a map depending on the request_inputs type
+    """
+
+    def __init__(self, *, config, request_inputs=None):
+        if request_inputs is None:
+            request_inputs = ""
+        self._config = config
+
+        if isinstance(request_inputs, dict):
+            self._body_data_map_interpolator = InterpolatedRequestInputProvider(config=config, request_inputs=request_inputs)
```

Comment:
could we pass down a `InterpolatedMapping` instead of an input provider?
It would simplify this class a bit since it shares the same interface as the string interpolator

Reply:
We'll just get rid of this class entirely in favor of `InterpolatedRequesTInputProvider`